skbio/io/util.py

```diff
@@ -183,17 +195,31 @@ def _munge_file(file, is_binary_file, arguments):
     errors = arguments.get("errors", _d["errors"])
     newline = arguments.get("newline", _d["newline"])
     compression = arguments.get("compression", _d["compression"])
+    container = arguments.get("container", _d["container"])
     is_output_binary = encoding == "binary"
     newfile = file
 
     compression_handler = get_compression_handler(compression)
+    container_handler = get_container_handler(container)
 
     if is_output_binary and newline is not _d["newline"]:
         raise ValueError("Cannot use `newline` with binary encoding.")
 
     if compression is not None and not compression_handler:
         raise ValueError("Unsupported compression: %r" % compression)
 
+    if container is not None and not container_handler:
+        raise ValueError("Unsupported container: %r" % container)
+
+    if container is not None:
+        if compression == "auto":
+            compression = None
+        elif compression is not None:
+            raise ValueError("Compression with a container is not supported.")
+
+    if container is not None and not is_binary_file:
+        raise ValueError("Containers are binary.")
```

Comment:
I think this is effectively true for container types we may care about. It is necessary to define in the documentation what we consider a container though as JSON _contains_ data. Perhaps the binary constraint can be relaxed in the future.